src/Symfony/Component/HttpClient/HttpOptions.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * A helper providing autocompletion for available options.
+ *
+ * @see HttpClientInterface for a description of each options.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class HttpOptions implements \IteratorAggregate
+{
+    private $options = [];
+
+    /**
+     * @return $this
+     */
+    public function setData($data)
```

Comment:
Why no return types in this class?

Reply:
Because no return types allow declaring fluent interfaces. `self` would be wrong and would give a false hint when using child classes. `static` is not possible of course. Better rely on the only accurate description: the annotation.

Reply:
Thank you for the explanation, didn’t know it